core/src/main/java/org/apache/carbondata/core/datamap/TableDataMap.java

```diff
@@ -135,7 +136,7 @@ public CarbonTable getTable() {
     int datamapsCount = 0;
     // In case if filter has matched partitions, then update the segments with datamap's
     // segment list, as getDataMaps will return segments that matches the partition.
-    if (null != partitions && !partitions.isEmpty()) {
+    if (isFilterPresent) {
```

Comment:
This reset of segments has to happen only when the partition and filter both are present.

Now in case of non-partition + just the filter also we are resetting.

Reply:
Now, we are loading cache for only the matched segments for filter (partition/non-partition). For non-partition table, based on segment level min-max. Hence we have to reset the segments.